server/routes/routerRequestHandlers.test.ts

```diff
@@ -32,71 +40,205 @@ describe('routerRequestHandlers', () => {
+
+        inductionService.getInduction.mockRejectedValue({
+          status: 404,
+          data: {
+            status: 404,
+            userMessage: `Induction not found for prisoner [${prisonNumber}]`,
+            developerMessage: `Induction not found for prisoner [${prisonNumber}]`,
+          },
+        })
+        const expectedError = createError(
+          404,
+          `Induction for Prisoner ${prisonNumber} not returned by the Induction API`,
+        )
+
+        // When
+        await requestHandler(
+          req as undefined as Request,
+          res as undefined as Response,
+          next as undefined as NextFunction,
+        )
+
+        // Then
+        expect(inductionService.getInduction).toHaveBeenCalledWith(prisonNumber, userToken)
+        expect(req.context.plan).toBeUndefined()
+        expect(next).toHaveBeenCalledWith(expectedError)
+      })
+
+      it('should call next function with error given retrieving Induction fails with a 500', async () => {
+        // Given
+        const prisonNumber = 'A1234BC'
+        req.params.id = prisonNumber
+
+        const userToken = 'a-user-token'
+        res.locals.user.token = userToken
+
+        inductionService.getInduction.mockRejectedValue({
+          status: 500,
+          data: {
+            status: 500,
+            userMessage: 'An unexpected error occurred',
+            developerMessage: 'An unexpected error occurred',
+          },
+        })
+        const expectedError = createError(
+          500,
+          `Induction for Prisoner ${prisonNumber} not returned by the Induction API`,
+        )
+
+        // When
+        await requestHandler(
+          req as undefined as Request,
+          res as undefined as Response,
+          next as undefined as NextFunction,
+        )
+
+        // Then
+        expect(inductionService.getInduction).toHaveBeenCalledWith(prisonNumber, userToken)
+        expect(req.context.plan).toBeUndefined()
+        expect(ciagPlanMapperMock).not.toHaveBeenCalled()
+        expect(next).toHaveBeenCalledWith(expectedError)
+      })
     })
 
-    it('should call next function with error given retrieving Induction fails with a 500', async () => {
-      // Given
-      const prisonNumber = 'A1234BC'
-      req.params.id = prisonNumber
-
-      const userToken = 'a-user-token'
-      res.locals.user.token = userToken
-
-      ciagService.getCiagPlan.mockRejectedValue(createError(500, 'Service unavailable'))
-      const expectedError = createError(500, `Induction for Prisoner ${prisonNumber} not returned by the Induction API`)
-
-      // When
-      await requestHandler(req as undefined as Request, res as undefined as Response, next as undefined as NextFunction)
-
-      // Then
-      expect(ciagService.getCiagPlan).toHaveBeenCalledWith(userToken, prisonNumber)
-      expect(req.context.plan).toBeUndefined()
-      expect(next).toHaveBeenCalledWith(expectedError)
+    // TODO - remove these tests in RR-623
```

Comment:
Overall this class looks like a massive diff. But its not really
What I've done is wrap the existing tests here in a describe, saying they use the CIAG API (based on the feature toggle in the beforeEach)

And above I've written the same tests (but with slightly different mock setup and expectations) in a describe block saying that its using the new Induction API

Reply:
RR-623 is the ticket on our backlog to remove the feature toggle 👍